### PR TITLE
Fix/docker version compatibility + docker image execution

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

**Comment by mzedel** [comment]
non buildkit using pipeline: https://gitlab.com/Northern.tech/Mender/mender-artifact/-/pipelines/617868677
buildkit using pipeline: https://gitlab.com/Northern.tech/Mender/mender-artifact/-/pipelines/617871038

**Comment by mzedel** [comment]
> @mender-test-bot, Let me know if you want to start the integration pipeline by mentioning me and the command "start pipeline".
> 
> my commands and options

🧱

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ include:
     file: '.gitlab-ci-check-commits.yml'
   - project: 'Northern.tech/Mender/mendertesting'
     file: '.gitlab-ci-check-license.yml'
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-check-docker-build.yml'
```
Comment (lluiscampos) [inline .gitlab-ci.yml:17]: Why including the template if we are overriding `build:docker` ?

File: Makefile [hunk 2]
```diff
@@ -58,13 +58,13 @@ build-natives:
 
 build-contained:
 	rm -f mender-artifact && \
-	image_id=$$(docker build -f Dockerfile . | awk '/Successfully built/{print $$NF;}') && \
+	image_id=$$(docker build -f Dockerfile -q .) && \
```
Comment (lluiscampos) [inline Makefile:61]: As the `Makefile` can be used outside of our CI, I think we should still support building without BUILDKIT. Can we check for it somehow in this command to then conditionally use `awk`?

Comment (mzedel) [reply Makefile:61]: => https://github.com/mendersoftware/mender-artifact/pull/414#issuecomment-1220657061 - this works for both...

File: .gitlab-ci.yml (hunk 1 above)
Comment (mzedel) [reply .gitlab-ci.yml:17]: argh sorry, that's a leftover - my intent was to use the standardized approach, however the `build:docker` job seems to not be used anywhere in the pipeline, I forgot to remove the template (preferably I would also remove the `build:docker` job, but I'm not sure if you might want to keep it to just test if that `make` target still works)

File: Makefile (hunk 2 above)
Comment (lluiscampos) [reply Makefile:61]: Sorry :facepalm:

File: .gitlab-ci.yml (hunk 1 above)
Comment (lluiscampos) [reply .gitlab-ci.yml:17]: While I see your good intentions, I think for this case is better to use the `Makefile` from the CI to guarantee we don't break it :+1:

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**